### ISSUE TITLE
A switch only can toggle/activate one item

### ISSUE THREAD

**Issue by relishrompgames** [TheGameCreators/GameGuruRepo#2764]
Can a switch only toggle one item? I have 3 lights and can only connect one to a switch. I would like the switch to turn on all of the lights.

VIDEO BELOW:
https://youtu.be/Fpvmv2dXsq0

[youtube]Fpvmv2dXsq0[/youtube]

**Comment by ArgentArts** [comment]
I can get multiple lights working on one switch without a problem:

![switch](https://user-images.githubusercontent.com/36735507/162062964-f563f99a-ea51-4d11-b03d-3ce7d3f4dba0.png)

https://user-images.githubusercontent.com/36735507/162063268-0181d7ba-3ca5-4e9e-9828-c43df24093c6.mp4

**Comment by LeeBamberTGC** [comment]
@relishrompgames This worked fine with latest EXP build, lights changed to toggle and off by default and then connected to a single switch, they all came on when switch thrown:

![image](https://user-images.githubusercontent.com/35840583/165818495-474e2140-f3c9-4ddf-8a64-550c93f0b80c.png)

**Comment by ArgentArts** [comment]
@relishrompgames Yeah, that's tricky with the visual logic nodes. The best bet with that is to place the door near the switch, connect them using the visual logic, then, after they are connected, move the door into position. It's not ideal, but it's better than trying a long distance connection. ;)

**Comment by LeeBamberTGC** [comment]
@ArgentArts Any suggestions on a second method to do this. I agree finding the second object to connect to can be super easy, but when they in hard to reach/find places the old IFUSED system was quicker.  Perhaps something like a dropdown which lists all objects in order of proximity, and select the target from there as a second method?

**Comment by ArgentArts** [comment]
There was a discussion about this in this topic:

https://github.com/TheGameCreators/GameGuruRepo/issues/1690

Perhaps some of the things there will help?

Now I am thinking outside the box:

What if the end-user could first select the Switch (now it's selected) then move about the map and find the object they want to connect it to and using a key combo (just like multi-select - CTRL+CLICK) they click the object to connect the Switch to (like a door or a light) and that makes the long-distance connection. This would only work when in the logic connection mode (i.e. Shooter, etc., is activated). The end-user could use the Detailed Object List to instantly zoom to the second object, too, so they don't have to "walk" through their map. So ...

1 - Activate Shooter mode
2 - Click on the Switch
3 - Zoom to the second, far away object (door, light, etc.)
4 - CTRL+Click to make the connection

DONE.

**Comment by ArgentArts** [comment]
@LeeBamberTGC So, you're saying they would have full-control of the cursor, just like as if they were not trying to connect any logic at all, as if they were still working on the level itself. If this were the case, and the end-user could use whatever tools the UI provides to get to the other object, then that works. Once they click on the second object, the connection is made.

The only thing that needs to be considered, I think, is ensuring that no accidental click along the way makes an accidental connection, forcing the end-user to have to go back to the switch, sever the connection, and start all over. So, perhaps a key-combo is required for the second object to ensure it's a unique click?
